.github/scripts/update-release-notes.js

```diff
@@ -0,0 +1,165 @@
+            str = new TextDecoder("utf-8").decode(buffer)
+            responseJson = JSON.parse(str)
+            
+            callback(responseJson.published_at, responseJson.body)
+            
+        });
+     
+    });
+     
+    reqGet.end();
+    reqGet.on('error', function(e) {
+        console.error(e);
+    });
+}
+
+
+function updateBOMReleaseNotes() {
+var optionsget = {
+    host : 'search.maven.org', 
+    port : 443,
+    path :  '/solrsearch/select?q=g:com.adobe.marketing.mobile+AND+a:sdk-bom&core=gav&rows=2&wt=json', 
+    method : 'GET',
+    headers: {
+        'Content-Type': 'application/json',
+        'User-Agent': 'server-side',
+      }
+};
+
+var reqGet = https.request(optionsget, function(res) {
+    // console.log("statusCode: ", res.statusCode);
+ 
+    res.on('data', function(d) {
+        str = new TextDecoder("utf-8").decode(d)
+        responseJson = JSON.parse(str)
+
+        responseJson.response.docs.forEach(element => {
+            if (isEarlierThan24Hours(element.timestamp) == false) {
+                if (element.a == 'sdk-bom') {
+                    const version = element.v
+
+                    fetchReleaseNotes('adobe','aepsdk-commons',`bom-${version}`, (ios1086DateStr, releaseNotes) => {
+                        const date = new Date(ios1086DateStr)
+                        const month = date.getMonth()
+                        const day = date.getDate()
+                        const year = date.getFullYear()
+                        const monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "June", "July", "Aug", "Sept", "Oct", "Nov", "Dec"];
+                        const shortMonth = monthNames[month]
+                        //## Aug 8, 2023
+                        console.error(`## ${shortMonth} ${day}, ${year}`);
+                        //### Android BOM 2.3.1
+                        console.error(`### Android BOM  ${version}`);
+                        console.error(releaseNotes)
+
+                        const releaseMdPath = "./src/pages/documentation/release-notes/index.md";
+
+                        const releaseNotesHeader = "# Release notes";
+                        // Read the contents of the markdown file.
+                        const contentArray = fs.readFileSync(releaseMdPath, "utf8").toString().split("\n");
+                         // Find the index of the release notes header.
+                        const releaseNotesHeaderIndex = contentArray.indexOf(releaseNotesHeader);
+                        console.error(releaseNotesHeaderIndex)
+
+                        if (releaseFileContainsLineStartWith(`### Android BOM  ${version}`) == true) {
+                            console.error("already updated")
+                            return
+                        }
+                        contentArray.splice(releaseNotesHeaderIndex + 1,0,"",`## ${shortMonth} ${day}, ${year}`,"",`### Android BOM  ${version}`, "",updateBOMReleaseNotesForAdobeIO(releaseNotes));
```

Comment:
You should also check if the date already exists and avoid inserting it.

Reply:
Yes, I do have code logic to check that. 
```javascript
if (releaseFileContainsLineStartWith(`### Android BOM  ${version}`) == true) {
                            console.error("already updated")
                            return
                        }
```

Reply:
There could still be other entries on the same date from manual PRs.